ui/lib/services/bom_bar_client.dart

```diff
@@ -38,9 +38,10 @@ class BomBarClient {
     ));
   }
 
-  /// Returns all available projects.
-  Future<List<Project>> getProjects() async {
-    final response = await dio.getUri(_projectsUrl);
+  /// Returns all or projects matching the provided [filter] fragment.
+  Future<List<Project>> findProjectsBySearchFragment(String? filter) async {
```

Comment:
I am missing a test for the `null` value of `filter`.

(My expectation is that this would request projects without passing a `q` parameter or perhaps an empty string.)